src/Tools/Source/RunTests/ProcessTestExecutor.cs

```diff
@@ -108,7 +108,7 @@ private async Task<TestResult> RunTestAsyncInternal(AssemblyInfo assemblyInfo, b
             try
             {
                 var commandLineArguments = GetCommandLineArguments(assemblyInfo);
-                var resultsFilePath = GetResultsFilePath(assemblyInfo);
+                var resultsFilePath = GetResultsFilePath(assemblyInfo, Options.IncludeHtml ? "html" : "xml");
```

Comment:
The `resultsFilePath` here should always be the XML path. Otherwise it breaks the logic below for looking up failed tests based on the XML contens of the results.

Reply:
Why do u think this change is necessary? The code on line 81 should include the HTML when the option is specified in build.ps1

Reply:
Without this change, we do pass the html flag to 'dotnet test' correctly, but we don't log the paths to the HTML logs from RunTests, and when tests complete with failures locally we open the XML logs instead of the HTML logs. #49091

Reply:
Perhaps resultsFilePath could be without an extension? Or in specific usages where we know Options.IncludeHtml is set we can try using ChangeExtension or something.

Reply:
Think the solution to that though is changing `TestResult.ResultsFilePath` more directly though. Yes this change accomplishes that but it does so but breaking the retry logic. Need to change that without breaking retry. 

Or possible just change `TestResult` to return both paths.

Reply:
I'll change this code since it definitely seems bad to just "sometimes try to parse HTML as XML", but I would also note that retry is disallowed when HTML output is enabled.

Reply:
Agreed HTML + retry is disallowed as a combo. It seems like a bad idea to make the rest of our code here aware of that though.